### ISSUE TITLE
Audit our lessons for where we should refer to independent and dependent variables

### ISSUE THREAD

**Issue by flannery-denny** [bootstrapworld/curriculum#1208]
- [x] Add a pedagogy box to scatterplots helping teachers understand why we use `explanatory` and `response` instead in this context. Here's the wording from 2021:

> We prefer the words “explanatory” and “response” in our curriculum, because in other contexts the words “dependent” and “independent” refer to whether or not the variables are related at all, as opposed to what role each plays in the relationship.

- [x] Look at the standards (at least NGSS, OK, CCmath) and think about where we could address their interest in students engaging with the terms independent and dependent by adding a sentence or two to lessons that are already addressing the concepts but not using the vocabulary
- [x] Add independent and dependent variable to the glossary
- [x] Update standards alignment

**Comment by flannery-denny** [comment]
@schanzer I've gathered the relevant standards in the comment above. Currently thinking:

- [ ]  perhaps the best way to address this issue is through #1279
- [ ] We probably aren't currently meeting the OK standards with the lessons we claim to

Thoughts?

**Comment by schanzer** [comment]
I like the idea of addressing this through #1279 , but it sounds like we should be using those terms in other lessons too (_especially_ contracts, but also player-animation, piecewise-functions-conditionals, functions-make-life-easier!). I think this would address the OK standards as well.

**Comment by flannery-denny** [comment]
CONTRACTS 
@schanzer I think if we really want to work independent and dependent variables into the contracts lesson we'd have to tackle it here, but I'm not sure I feel great about it because the functions being presented in this lesson are not necessarily of the kind that take in an independent variable and produce a dependent variable...

> In programming, our verbs are called @vocab{functions}. A function is like a machine, and it does three things:
> 
> - It takes in some values (called @vocab{arguments})
> - It _does something_ to those values
> - It produces a new value

perhaps it could read something like the following?

> In programming, our verbs are called @vocab{functions}. A function is like a machine, and it does three things:
> 
> - It takes in some values (called @vocab{arguments}) for the independent variable
> - It _does something_ to those values
> - It produces a new value for the dependent variable

Or perhaps we'd leave the initial bullets alone and add a sentence that reads something like

There are lots of functions that take in numbers and use them to produce numbers.  For example, we might have a functions that calculates the amount of money earned based on the number of hours worked. We can describe the time being taken in as the independent variable and the money being earned as the dependent variable, because the amount of money we will earn depends on how many hours we babysit for.

But, honestly, this feels more relevant to our lessons focused on linear functions... and like it might make this lesson more overwhelming. I'll look at each of the other lessons shortly.

**Comment by schanzer** [comment]
The premise of a function is that the output depends _entirely_ on the input - and nothing else. This is a corollary to the vertical line test, which says that the same input will always produce the same output. If the output was independent from the inputs in any way, the function would fail the test.

@flannery-denny You write:

> the functions being presented in this lesson are not necessarily of the kind that take in an independent variable and produce a dependent variable...

I don't understand what you mean here. How are these any different from any of the other functions that you describe later on (e.g. -  "functions that calculates the amount of money earned based on the number of hours worked"). All of them pass the vertical line test, and all of them have outputs that depend entirely on their inputs.

**Comment by flannery-denny** [comment]
@schanzer 
please see the comment above re: standards and let me know 
- [ ] is it ok to add all bolded lesson additions
- [ ] is it ok to remove all italicized lesson deletions
- [ ] how would you like to proceed re: satisfying standards by making using the words independent and dependent variables in scatterplots and/or lr?

Also, please have a look at the commit adding independent and dependent variables to function-definitions-linear and 
- [ ] confirm that it works for you.

If there is an item we should discuss, please add it to our Tuesday agenda. Thanks!

**Comment by schanzer** [comment]
@flannery-denny apologies for the delay. 

- Adding to bolded lessons makes sense. Go for it.
- I still feel like we're missing an opportunity _not_ talking about this in Contracts. I'm cool with you removing from the other italicized lessons, but let's talk more about Contracts
- Talking about them in LR makes total sense, since the _predictor function_ is defined in terms of dependent and independent. I think they can be added to scatter plots by way of counterexample: "why don't we use the terms dependent and independent here?"
- The commit looks good

**Comment by flannery-denny** [comment]
Add "The domain of a function is all of the independent variables that govern the output" to contracts lesson plan.

**Comment by flannery-denny** [comment]
@schanzer at present we introduce the word Domain here in the lesson plan

> . *Name* -- the name of the function, which we type in whenever we want to use it
> . *Domain* -- the type(s) of data we give to the function. 
> . *Range* -- the type of data the function produces

In the glossary, we define Domain as

>  the type or set of inputs that a function expects

Confirming that you are comfortable with one or both of these definition shifting its focus away from types so that we can incorporate the following wording to the lesson

> "The domain of a function is all of the independent variables that govern the output"

**Comment by schanzer** [comment]
@flannery-denny I'd combine them: "the type or set of inputs that a function expects. These are the independent variables which govern the output of the function"

**Comment by flannery-denny** [comment]
@schanzer I've update the definitions for domain and range in the glossary to include the words independent and dependent variables.

I've reread the Contracts lesson, and, other than those updates to the glossary, the only mention of the word variable is as follows: 

> Note that arguments (or "inputs") are the values passed into a function. This is different from variables, which are the placeholders that get replaced with those arguments! Pyret has lots of built-in functions, which we can use to write more interesting programs.

I still don't believe that the contracts lesson as it is currently written satisfies the standard below 

> "OK.PA.A.1.1": {
> "description": "Recognize that a function is a relationship between an independent variable and a dependent variable in which the value of the independent variable determines the value of the dependent variable.",

but I have left it as satisfied in the dictionary

**Comment by flannery-denny** [comment]
@schanzer I'm realizing that this kind of wording around standards from im would eliminate our occasional disagreements about which lessons connect to what. The way our stuff is written I have a sense of finality... like we really have to be getting to the standard. You sometimes want to include things that are building toward an outcome.
<img width="239" alt="Screen Shot 2023-06-14 at 3 43 29 PM" src="https://github.com/bootstrapworld/curriculum/assets/66691020/2dc4a9d5-736d-4a02-8477-d1fc9ed587aa">

**Comment by schanzer** [comment]
@flannery-denny *smiling* everything old is new again. Kathi and I used this exact distinction in the pre-Dorai build system.

Unfortunately, this wound up making the standards alignment incredibly complicated to keep updated. In practice, stuff just fell out of date quickly. I'm nervous about going back to that as it would mean an enormous investment of time to get started and then a larger investment of time to maintain...and I'm just not convinced it's worth the investment for a small team like ours.

I've added this as a discussion topic to our curriculum agenda.
